topostats/io.py

```diff
@@ -1284,3 +1284,369 @@ def dict_to_json(data: dict, output_dir: str | Path, filename: str | Path, inden
     output_file = output_dir / filename
     with output_file.open("w") as f:
         json.dump(data, f, indent=indent, cls=NumpyEncoder)
+
+
+class TopoFileHelper:
+    """
+    Helper class for searching through the data in a .topostats (hdf5) file.
+
+    Parameters
+    ----------
+    topofile : Path
+        Path to the .topostats file.
+
+    Examples
+    --------
+    Creating a helper object.
+    ```python
```

Comment:
Not sure about how this renders when Sphinx's Autoapi-doc parses it to generate API docs in the webpage as docstrings are Restructured text. Might be worth using the `.. code-block::` approach, see @MaxGamill-Sheffield solution in commit b953634e6c98156d6d26e35518302c2167b9739e .

Reply:
Very good point thank you, I'll do that